src/libraries/Native/Unix/System.Native/pal_signal.c

```diff
@@ -17,33 +17,179 @@
 #include <unistd.h>
 
 static pthread_mutex_t lock = PTHREAD_MUTEX_INITIALIZER;
-static struct sigaction g_origSigIntHandler, g_origSigQuitHandler; // saved signal handlers for ctrl handling
-static struct sigaction g_origSigContHandler, g_origSigChldHandler; // saved signal handlers for reinitialization
-static struct sigaction g_origSigWinchHandler; // saved signal handlers for SIGWINCH
-static volatile CtrlCallback g_ctrlCallback = NULL; // Callback invoked for SIGINT/SIGQUIT
-static volatile TerminalInvalidationCallback g_terminalInvalidationCallback = NULL; // Callback invoked for SIGCHLD/SIGCONT/SIGWINCH
-static volatile SigChldCallback g_sigChldCallback = NULL; // Callback invoked for SIGCHLD
+
+// Saved signal handlers
+static struct sigaction g_origSigHandler[NSIG];
```

Comment:
Are we sure that the `NSIG` macro includes real time signals on all platforms?

Reply:
Apparently it does not on BSD: https://bugs.python.org/issue20584.